Controller/Adminhtml/AbstractMenu.php

```diff
@@ -0,0 +1,36 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Snowdog\Menu\Controller\Adminhtml;
+
+use Magento\Backend\App\Action;
+use Magento\Backend\Model\View\Result\Page;
+
+abstract class AbstractMenu extends Action
+{
+    /**
+     * Authorization level of a basic admin session
+     *
+     * @see _isAllowed()
+     */
+    const ADMIN_RESOURCE = 'Snowdog_Menu::menus';
+
+    /**
+     * Menu ID key
+     */
+    const ID = 'menu_id';
```

Comment:
Is it really necessary?

Reply:
Yes, it is necessary. We will use it in child classes where we set `id` instead of `menu_id`.